src/LieTheory/WeylGroup.jl

```diff
@@ -753,6 +753,21 @@ function reflection(beta::RootSpaceElem)
   return W([list_of_indices; current_index; reverse(list_of_indices)])
 end
 
+###############################################################################
+# G-set functionality
+
+# TODO: Better way to do this, can be inefficient.
+function action_homomorphism(Omega::GSetByElements{WeylGroup})
```

Comment:
This function looks semantically wrong to me. The problem is that you do the following:
You have a group `W` that acts on a G-Set `Omega`. You now take a group `G` that is isomorphic to `W` and *take the action on itself*. Here, you would need to instead translate to the action of `G` on `Omega`, if you want to delegate to the G-Set functionality for PermGroups.

Reply:
Yes.

I have discussed with @jamesnohilly how we can deal with this situation:
Compute an isomorphic permutation group `P` and let it act on the given set like `W` acts on it. Then construct an action homomorphism for `P` w.r.t. this action. The desired action homomorphism is the composition of the two maps.